libsyclinterface/CMakeLists.txt

```diff
@@ -156,17 +161,49 @@ add_library(DPCTLSyclInterface
     ${helper_sources}
 )
 
-target_include_directories(DPCTLSyclInterface
-    PRIVATE
-    ${CMAKE_CURRENT_SOURCE_DIR}/include/
-    ${CMAKE_CURRENT_SOURCE_DIR}/helper/include/
-    ${IntelSycl_SYCL_INCLUDE_DIR}
-)
+if(DPCTL_ENABLE_GLOG)
+    find_path(GLOG_INCLUDE_DIR NAMES glog/logging.h)
+    find_library(GLOG_LIBRARY NAMES glog)
+    if(GLOG_INCLUDE_DIR AND GLOG_LIBRARY)
+        message(STATUS "Found GLog:")
+        message(STATUS " Headers: ${GLOG_INCLUDE_DIR}")
+        message(STATUS " Library: ${GLOG_LIBRARY}")
+    else()
+        message(FATAL_ERROR "GLog not found")
+    endif()
+
+    find_package(glog REQUIRED)
+
+    target_include_directories(DPCTLSyclInterface
+        PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/include/
+        ${CMAKE_CURRENT_SOURCE_DIR}/helper/include/
+        ${IntelSycl_SYCL_INCLUDE_DIR}
+        ${GLOG_INCLUDE_DIR}
+    )
+
+    target_compile_definitions(DPCTLSyclInterface PUBLIC ENABLE_GLOG)
+
+    target_link_libraries(DPCTLSyclInterface
+        PRIVATE ${IntelSycl_SYCL_LIBRARY}
+        PRIVATE ${IntelSycl_OPENCL_LIBRARY}
+        ${GLOG_LIBRARY}
+    )
+
+else()
+    target_include_directories(DPCTLSyclInterface
```

Comment:
There is no need to duplicate `target_link_libraries` in both branches. You can do:

``` cmake
target_include_directories(DPCTLSyclInterface
    PRIVATE
    ${CMAKE_CURRENT_SOURCE_DIR}/include/
    ${CMAKE_CURRENT_SOURCE_DIR}/helper/include/
    ${IntelSycl_SYCL_INCLUDE_DIR}
)
target_link_libraries(DPCTLSyclInterface
    PRIVATE ${IntelSycl_SYCL_LIBRARY}
    PRIVATE ${IntelSycl_OPENCL_LIBRARY}
)
if(DPCTL_ENABLE_GLOG)
   # logic for finding GLOB skipped
    target_include_directories(DPCTLSyclInterface
        PRIVATE
        ${GLOG_INCLUDE_DIR}
    )
   target_compile_definitions(DPCTLSyclInterface PRIVATE ENABLE_GLOG)
   target_link_libraries(DPCTLSyclInterface
        PRIVATE ${GLOG_LIBRARY}
    )
endif()
```

Reply:
change pushed